src/Opdex.Platform.WebApi/IndexerBackgroundService.cs

```diff
@@ -120,18 +114,31 @@ protected override async Task ExecuteAsync(CancellationToken cancellationToken)
                 }
                 finally
                 {
-                    started = true;
+                    // delay 30 seconds when indexing services are unavailable
+                    // delay 8 to 12 seconds when indexing is available
+                    var seconds = unavailable ? 30 : new Random().Next(8, 13);
+                    await Task.Delay(TimeSpan.FromSeconds(seconds), cancellationToken);
```

Comment:
This can affect shutdowns but idk if its a huge deal since the important operations should have finished.

We allow 30 second timeout on shutdown, really that may be the problem when taking polly + retries into consideration, maybe it should be more like several minutes to make sure shutdown isn't forced in the middle of anything.

I don't think this code particularly is problematic but the comment is relevant

Reply:
Maybe a simple fix for this specifically actually
```
if (!cancellationToken.IsCancellationRequested) { ... } 
```

Reply:
Not sure I understand. As we're passing the cancellation token into `Task.Delay` I'm not sure there is a difference, as the cancellation will throw `TaskCancelledException` and the delay is over. The indexer *should* be locked at this time anyway.

As a side note, I thought about shutdown and the cancellation token. Inside `StopAsync` I've now moved the call to `base.StopAsync` up to the top, as this is what requests cancellation of the token. My thinking is we are better off cancelling any potential indexing before force unlocking the indexer.

Reply:
We want to do everything we can to shut down gracefully. The snippet I sent will not hit Task.Delay if we wrap it in that if block preventing unwanted exceptions from being thrown.

I think its better if we move the stop back into the the finally block of that try catch, basically to make sure we've finished the attempt prior to actually calling base.Stop. Imo its better to order that way even for a slight advantage rather than the stopping first potentially running into issues related with the attempted check and unlock.

Reply:
I've reverted the `Task.Delay` so it's now called inside the try block with the started flag enabled. Also added a catch block for `TaskCancelledException` so we can log something more specific. 

As for the `StopAsync()` call, I don't understand what issues could potentially occur. I believe calling `base.StopAsync()` after attempting to unlock opens up a slim possibility of the indexer being unlocked then locked once again, as the cancellation token is not cancelled until `base.StopAsync()` is called. However, this is covered by the try-finally block that the call to process latest blocks is wrapped in, so I'm happy to revert it back as well.

Reply:
I suppose `base.StopAsync()` could be called first. Either way, theres a slight chance of edge cases for us if we're doing anything related to locking/unlocking.

I think the changes you made look good and if you feel that putting `StopAsync` first is the better move, do it 👍